subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/PropertyValidationAccess.java

```diff
@@ -93,13 +94,16 @@ public void createAndAddToQueue(BeanTypeNode<?> parentNode, String propertyName,
         private BeanTypeNode<?> create(@Nullable BeanTypeNode<?> parentNode, @Nullable String propertyName, TypeToken<?> beanType) {
             Class<?> rawType = beanType.getRawType();
             TypeMetadata typeMetadata = metadataStore.getTypeMetadata(rawType);
-            if (propertyName != null && !typeMetadata.hasAnnotatedProperties()) {
+            if (parentNode != null && propertyName != null && !typeMetadata.hasAnnotatedProperties()) {
```

Comment:
We don't (they are either both null or both non-null). If I only do a null check on one of the two properties I get a nullable further down when using the value.

Reply:
But can either be `null`, ever? That's only true for root nodes, but root nodes are created via `createRootNode()`. I think what we should do is to remove the `@Nullable` from the parameters, because this method is never called with `null` parameters.

Reply:
We could even emphasize this by renaming `create()` to `createChild()`. I love how all these nullability checks let us simplify code.

Reply:
I removed the checks.

Reply:
Was worth waking up that day.